src/wdio/services/compare/LocalScreenshotProcessor.js

```diff
@@ -0,0 +1,106 @@
+import { terraViewports as VIEWPORTS } from '../../../../config/wdio/services.default-config';
+
+const screenshotSetup = {
+  diffDir: 'diff',
+  referenceDir: 'reference',
+  screenshotDir: 'latest',
+};
+
+const testIdRegex = /\[([^)]+)\]/;
+
+export default class LocalScreenshotProcessor extends TerraCompare {
+  static createTestName(fullName) {
+    const matches = testIdRegex.exec(fullName);
+
+    // If test ID is provided, use the ID for a shorter test name, otherwise use the full name
+    let name = matches ? matches[1] : fullName.trim();
+
+    // Remove white space
+    name = name.replace(/[\s+.]/g, '_');
+
+    // Remove windows reserved characters. See: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#naming_conventions
+    // eslint-disable-next-line no-useless-escape
+    name = name.replace(/[\?\<\>\/\\|\*\"\:\+\.]/g, '-');
+
+    return name;
+  }
+
+  static getScreenshotName(context) {
+    const parentName = LocalScreenshotProcessor.createTestName(context.test.parent);
+    const testName = LocalScreenshotProcessor.createTestName(context.test.title);
+
+    return `${parentName}[${testName}].png`;
+  }
+
+  static getFormFactor(context) {
+    let formFactor = global.browser.options.formFactor;
+
+    if (!formFactor) {
+      const browserWidth = context.meta.viewport.width;
+      const viewportSizes = Object.keys(VIEWPORTS);
+      viewportSizes.some((viewport) => {
+        if (browserWidth <= VIEWPORTS[viewport].width) {
+          formFactor = VIEWPORTS[viewport].name;
+          return true;
+        }
+        return false;
+      });
+    }
+
+    // If viewport value is undefined due to being larger then the terra defiend viewports, set formFactor to enormous.
+    formFactor = formFactor || 'enormous';
+
+    return formFactor;
+  }
+
+  static getScreenshotDir(context) {
+    const locale = global.browser.options.locale || 'en';
+    const browserName = context.desiredCapabilities.browserName;
+    const formFactor = LocalScreenshotProcessor.getFormFactor(context);
+    const testForm = `${browserName}_${formFactor}`;
+    const testSuite = path.parse(context.test.file).name;
+
+    return path.join(locale, testForm, testSuite);
+  }
+
+  static getScreenshotPath(ref, context) {
+    let testPath = path.dirname(context.test.file);
+
+    const baseDir = global.browser.options.baseScreenshotDir;
+    if (baseDir) {
+      testPath = testPath.split(process.cwd())[1];
+      // Added to allow for test reusablility from terra repositories
+      if (testPath.includes('node_modules')) {
+        testPath = testPath.split('node_modules')[1];
+      }
+      testPath = baseDir + testPath;
+    }
+    const refDir = screenshotSetup[`${ref}Dir`];
+
+    return path.join(testPath, '__snapshots__', refDir, LocalScreenshotProcessor.getScreenshotDir(context), LocalScreenshotProcessor.getScreenshotName(context));
+  }
+
+  static async getReferenceScreenshotData(context) {
```

Comment:
Talked offline but adding here to I/we don't forget.

Not sure with how this would be called from webdriver.io offhand, but it might be worth checking if we can return the reference directory, instead of the reference screenshot, and then filter on what screenshots were actually compared against to find 'stale' screenshots

Reply:
I'd like to do this at some point, but I think this may be out of scope for this right now.